app/src/main/java/helium314/keyboard/keyboard/KeyboardLayoutSet.java

```diff
@@ -70,6 +70,8 @@ public KeyboardLayoutSetException(final Throwable cause, final KeyboardId keyboa
         }
     }
 
+    public record InternalAction(int code, String label) {}
```

Comment:
Could you add some description to make clear what it does?
E.g. _key code and label overriding the action provided by the input field_.